mrp_partial_production/models/mrp.py

```diff
@@ -62,60 +38,19 @@ def get_qty_available_to_produce(self):
         """Compute the total available to produce considering
         the lines reserved
         """
-        uom_obj = self.env['product.uom']
+        qty = []
         for record in self:
-            bom_obj = self.env['mrp.bom']
-            if not record.move_lines.mapped('reserved_quant_ids'):
-                return 0.0
-            self._cr.execute('''
-                             SELECT m.product_id,
-                                    sum(q.qty) AS total
-                             FROM stock_move AS m
-                             LEFT OUTER JOIN stock_quant AS q ON
-                                             q.reservation_id=m.id
-                             WHERE m.raw_material_production_id = {prod_id}
-                             GROUP BY m.product_id;
-                             '''.format(prod_id=record.id))
-            result = {i.get('product_id'): i.get('total') or 0
-                      for i in self._cr.dictfetchall()}
-            product_uom_qty = uom_obj._compute_qty(record.product_uom.id,
-                                                   1,
-                                                   record.product_id.uom_id.id)
-            consume_lines = bom_obj._bom_explode(record.bom_id,
-                                                 record.product_id,
-                                                 product_uom_qty)
-
-            qty = []
-            for line in consume_lines and consume_lines[0] or ():
-                product_id = line.get('product_id')
-                val = line.get('product_qty') and \
-                    int(result.get(product_id, 0) /
-                        line.get('product_qty')) or 0
+            result = {line.product_id.id: line.reserved_availability
```

Comment:
what about use mapped?

Reply:
How can I do using mapped to obtain a dict like {id: value}?

Reply:
@ysantiago @JulioSerna as move_raw_ids can provide duplicated ids for product_id, result is not providing a proper accumulation of reserved_availability.

Please fix this.

<img width="599" alt="addons-vauxoo_mrp_py_at_b376543158a6cf818f1cfa93ffeb0f013d482923_ _vauxoo-dev_addons-vauxoo" src="https://user-images.githubusercontent.com/7598010/30281733-33f2d14c-96e1-11e7-85a1-d95ea06a1606.png">

Reply:
@hbto @JulioSerna in commit [dee48ed](https://github.com/Vauxoo/addons-vauxoo/pull/1227/commits/dee48ede06ad2c982a44b685e4043cffc30aa63b) is the change

Reply:
Before
![photo_2017-09-12_11-26-07](https://user-images.githubusercontent.com/11341058/30334113-a7f5fec2-97ac-11e7-9b82-327bc36893ed.jpg)


After
![photo_2017-09-12_11-23-46](https://user-images.githubusercontent.com/11341058/30334051-7a858d90-97ac-11e7-8c0e-18ef5f768e43.jpg)